packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,550 @@
+            signInController.selectAuthenticator(authenticator: selectedOption,
+                                                 completion: completion)
+        }
+    }
+}
+
+```
+
+These screenshots show selecting Google Authenticator using the form. The first displays the picker with the label and no selected item. The next screen shows the generated list of choices shown after tapping the picker. The final screen shows the picker with Google Authenticator selected.
+
+<div class="common-image-format">
+
+![The authentication selection picker form.](/img/authenticators/ios-authenticators-google-choose-authenticator.png "A series of three screenshots with a form for selecting an authenticator. The first shows the picker with no value, the second shows the list of choices in a new form, and the third shows the picker displaying the name of the chosen authenticator.")
+
+</div>
+
+
+The UI updates the sign-on controller with the selected authenticator by calling `selectAuthenticator(authenticator:completion)`, or `skipAuthenticatorSelection(completion:)` if the user choses to skip enrolling in additional authenticators.
+
+```swift
+public func skipAuthenticatorSelection(completion:@escaping (Result<SuccessResultType, LoginError>) -> Void) {
+    // Set the current view model completion handler which is
+    // called when recieving a response from the SDK.
+    self.completion = completion
+
+    guard let skipRemediation = currentResponse?.remediations.filter({ $0.type == .skip }).first else {
+        // Handle the error
+        return
+    }
+
+    // Request the next step.
+    skipRemediation.proceed()
+}
+
+
+public func selectAuthenticator(authenticator: Remediation.Form.Field?, completion:@escaping (Result<SuccessResultType, LoginError>) -> Void) {
+    // Set the selected authenticator.
+    self.completion = completion
+
+    guard let selectorFields = authenticatorOptions() else {
+              self.finish(with: .cannotProceed)
+              return
+          }
+
+    // Update the remediation and send it to the server,
+    // requesting the next step.
+    selectorFields.optionSelectorField.selectedOption = authenticator
+    currentRemediation!.proceed()
+}
+```
+
+### 5: Display the shared secret, QR Code, and request the code
+
+After selecting the authenticator, the server sends an enrollment request in a remediation of type `.enrollAuthenticator`. Enrolling uses a secret key shared between Okta and Google Authenticator that's used to generate and verify the generated time-based one-time passcodes (TOTP).
+
+The remediation includes the shared secret as both a QR Code and as text. It also includes a request to enter a Google Authenticator code after enrollment.
+
+The sign-on controller code for `handleEnrollAuthenticator(remediation:)` works for many different authenticators because the `form` property of the remediation contains the information required to build the UI:
```

Comment:
```suggestion
The sign-in controller code for `handleEnrollAuthenticator(remediation:)` works for many different authenticators because the `form` property of the remediation contains the information required to build the UI:
```

Reply:
Fixed in the next commit.